internal/db/models/task.go

```
@@ -0,0 +1,72 @@
+	case string(TaskStatusRunning):
+		return TaskStatusRunning, nil
+	case string(TaskStatusCompleted):
+		return TaskStatusCompleted, nil
+	case string(TaskStatusFailed):
+		return TaskStatusFailed, nil
+	default:
+		return TaskStatusUnknown, fmt.Errorf("invalid task status: %s", str)
+	}
+}
+
+func (s *TaskStatus) UnmarshalJSON(data []byte) error {
```

golangci-lint (1.64.8)

[warning] exported: exported method TaskStatus.UnmarshalJSON should have comment or be unexported

(revive)